integration/spark/src/main/common/java/io/openlineage/spark/api/DatasetFactory.java

```diff
@@ -0,0 +1,109 @@
+package io.openlineage.spark.api;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.DatasetIdentifier;
+import io.openlineage.spark.agent.util.PlanUtils;
+import java.net.URI;
+import org.apache.spark.sql.types.StructType;
+
+/** @param <D> */
```

Comment:
_Minor_: I would add a description for `D` and some javadocs for the class ;)

Reply:
I swear I wrote these javadocs, but... they vanished... 😅